snappy.cc

```diff
@@ -721,8 +784,8 @@ static inline uint32 ExtractLowBytes(uint32 v, int n) {
   return _bzhi_u32(v, 8 * n);
 #else
   // This needs to be wider than uint32 otherwise `mask << 32` will be
-  // undefined.
-  uint64 mask = 0xffffffff;
+  // undefined. 
+  uint64 mask = ~0;
   return v & ~(mask << (8 * n));
```

Comment:
@dmiller423 You are changing here existent code, which introduces review effort. Are you absolutely(!) sure that this is necessary? If so, the add a comment line, possibly with a link to high-quality documentation. Otherwise just reverse the change.